File: backend/app/models/ticket_config.py
from sqlalchemy import Column, Integer, String, Numeric, ForeignKey
from sqlalchemy.orm import relationship
from app.db.base import Base


class TicketConfig(Base):
    """Defines ticket types, prices and quantities for an EventSchedule.
    Price is fixed per venue (schedule), not per day."""
    __tablename__ = "ticket_configs"
    config_id = Column(Integer, primary_key=True, index=True)
    schedule_id = Column(Integer, ForeignKey("event_schedules.schedule_id", ondelete="CASCADE"), nullable=False, index=True)
    ticket_type = Column(String(100), nullable=False)
    price = Column(Numeric(12, 2), nullable=False)
    max_quantity = Column(Integer, nullable=False)

    schedule = relationship("EventSchedule", back_populates="ticket_configs")
    e_tickets = relationship("ETicket", back_populates="ticket_config", cascade="all, delete-orphan")

    @property
    def remaining_quantity(self):
        return None

    @property
    def created_at(self):
        return None

    @property
    def updated_at(self):
        return None

File: frontend/src/types/ticketConfig.ts
export type TicketConfig = {
  config_id: number;
  schedule_id: number;
  ticket_type: string;
  price: string | number;
  max_quantity: number;
  remaining_quantity?: number | null;
  created_at: string;
  updated_at: string;
};

export type TicketConfigCreate = {
  schedule_id: number;
  ticket_type: string;
  price: number;
  max_quantity: number;
};

